File: SeleniumExercise/src/test/java/org/vena/uitest/InterviewTest.java
package org.vena.uitest;

import applicationpages.manager.CreateModalPage;
import applicationpages.manager.ManagerPage;
import io.github.bonigarcia.wdm.WebDriverManager;
import java.util.concurrent.TimeUnit;
import org.openqa.selenium.By;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.vena.BaseTest;
import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertNotNull;
import static org.testng.AssertJUnit.assertTrue;

public class InterviewTest extends BaseTest {
	private WebDriver driver;

	private static ManagerPage managerPage;

	@BeforeClass
	public void setUp() {
		WebDriverManager.chromedriver().setup();

		driver = new ChromeDriver();
		driver.manage().window().setSize(new Dimension(1920, 1080));
	}

	@Test(priority = 1)
	public void testLogin() {
		driver.get("https://dev.vena.io");
		driver.findElement(By.id("email")).sendKeys(credentials.getUsername());
		driver.findElement(By.id("password")).sendKeys(credentials.getPassword());
		driver.findElement(By.cssSelector("[data-testid*='login-pw']")).click();
		driver.manage().timeouts().implicitlyWait(30L, TimeUnit.MILLISECONDS);
		assertNotNull(driver.findElement(By.cssSelector("[name='manager']")));
		String userName = driver.findElement(By.id("accountTab")).getText();
		assertEquals(userName, "Admin User");
	}

	@Test(priority = 2)
	public void testCreateFolder() {
		managerPage = new ManagerPage(driver);
		managerPage.clickCreateButton();
		CreateModalPage createModalPage = managerPage.selectFolderOption();
		createModalPage.setName("Test New Folder");
		managerPage = createModalPage.clickCreateButton();
		assertTrue(managerPage.getFoldersList().contains("Test New Folder"));
	}

	@Test(priority = 3)
	public void testCreateProcessUnderFolder() throws InterruptedException {
		managerPage = new ManagerPage(driver);
		managerPage.clickCreateButton();
		managerPage.selectFolderOption();
		WebElement nameTextBox = driver.findElement(By.cssSelector("[data-testid='inputDialog-TextField'] div input"));
		nameTextBox.sendKeys("Test New Folder 1");
		WebElement createButtonOnModal = driver.findElement(By.cssSelector("[data-testid='submit-Create']"));
		createButtonOnModal.click();
		Thread.sleep(5000);
		managerPage.clickFolder("Test New Folder 1");
		managerPage.clickCreateButton();
		managerPage.selectProcessOption();
		nameTextBox.sendKeys("Test New Process 1");
		createButtonOnModal.click();
	}

	@Test(priority = 4)
	public void testCreateButtonDisabled() {
		managerPage = new ManagerPage(driver);
		managerPage.clickCreateButton();
		CreateModalPage createModalPage = managerPage.selectFolderOption();
		assertTrue(createModalPage.isCreateButtonEnabled());
	}

	@Test(priority = 5)
	public void testClickProcess() {
		managerPage = new ManagerPage(driver);
		managerPage.clickProcess("Test New Process 1");
	}

	@AfterClass (alwaysRun = true)
	public void teardown() {
		driver.quit();
		deleteAllCustomFolders();
	}
}

File: SeleniumExerciseC#/Test/uitest/InterviewTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using WebDriverManager;
using WebDriverManager.DriverConfigs.Impl;
using Pages;

namespace Test
{
    [TestClass]
    public class InterviewTest : BaseTest
    {
        private static ManagerPage _managerPage { get; set; } = default!;
        private static IWebDriver _driver { get; set; } = default!;

        [ClassInitialize]
        static public void SetUp(TestContext context)
        {
            new DriverManager().SetUpDriver(new ChromeConfig());
            _driver = new ChromeDriver();
            _driver.Manage().Window.Maximize();
        }

        [TestMethod]
        public void test1_testLogin()
        {
            _driver.Navigate().GoToUrl("https://dev.vena.io");
            _driver.FindElement(By.Id("email")).SendKeys(credentials.username);
            _driver.FindElement(By.Id("password")).SendKeys(credentials.password);
            _driver.FindElement(By.CssSelector("[data-testid*='login-pw']")).Click();
            _driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromMilliseconds(30);
            Assert.IsNotNull(_driver.FindElement(By.CssSelector("[name='manager']")));
            String userName = _driver.FindElement(By.Id("accountTab")).Text;
            Assert.AreEqual(userName, "Admin User");
        }

        [TestMethod]
        public void test2_testCreateFolder()
        {
            _managerPage = new ManagerPage(_driver);
            _managerPage.clickCreateButton();
            CreateModalPage createModalPage = _managerPage.selectFolderOption();
            createModalPage.setName("Test New Folder");
            _managerPage = createModalPage.clickCreateButton();
            Assert.IsTrue(_managerPage.getFoldersList().Contains("Test New Folder"));
        }

        [TestMethod]
        public void test3_testCreateProcessUnderFolder()
        {
            _managerPage = new ManagerPage(_driver);
            _managerPage.clickCreateButton();
            _managerPage.selectFolderOption();
            IWebElement nameTextBox = _driver.FindElement(By.CssSelector("[data-testid='inputDialog-TextField'] div input"));
            nameTextBox.SendKeys("Test New Folder 1");
            IWebElement createButtonOnModal = _driver.FindElement(By.CssSelector("[data-testid='submit-Create']"));
            createButtonOnModal.Click();
            Thread.Sleep(5000);
            _managerPage.clickFolder("Test New Folder 1");
            _managerPage.clickCreateButton();
            _managerPage.selectProcessOption();
            nameTextBox.SendKeys("Test New Process 1");
            createButtonOnModal.Click();
        }

        [TestMethod]
        public void test4_testCreateButtonDisabled()
        {
            _managerPage = new ManagerPage(_driver);
            _managerPage.clickCreateButton();
            CreateModalPage createModalPage = _managerPage.selectFolderOption();
            Assert.IsTrue(createModalPage.isCreateButtonEnabled());
        }

        [TestMethod]
        public void test5_testClickProcess()
        {
            _managerPage = new ManagerPage(_driver);
            _managerPage.clickProcess("Test New Process 1");
        }

        [ClassCleanup]
        public static void TearDown()
        {
            _driver.Quit();
            BaseTest.deleteAllCustomFolders();
        }
    }
}